### PR TITLE
Address null ref when we can't get an Undo manager.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Mitigates: https://github.com/dotnet/roslyn/issues/17898

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

**Comment by jasonmalinowski** [comment]
Also, why are we getting an undo manager in the first place? Something seems fishy here.

**Comment by CyrusNajmabadi** [comment]
> Also, why are we getting an undo manager in the first place? Something seems fishy here.

We're getting an undo manager so we can undo the 'move type' operation.

**Comment by CyrusNajmabadi** [comment]
> Can we split up the formatting changes into separate commits?

Sure.

**Comment by CyrusNajmabadi** [comment]
@jasonmalinowski Have chnaged this into two commits.

**Comment by jasonmalinowski** [comment]
@CyrusNajmabadi But then how did this ever work before? Do we not have test coverage for it?

**Comment by CyrusNajmabadi** [comment]
it works fine normally.  i'm just being resilient to it failing for... reasons...

**Comment by jasonmalinowski** [comment]
So why didn't it work fine sometimes? Is this a race?

**Comment by CyrusNajmabadi** [comment]
@jasonmalinowski What are your thoughts on this.  Dave doesn't seem to have any reliable sort of repro.  My running theory is that it's a race, whereby we are referring to a document id that isn't backed by a Document anymore due to the solution changing under us.  I'm not sure what is the right thing to do in such a situation.

Should i just keep the code as is, and we accept the crash until we figure it out?  Or would it be better to be resilient here?

**Comment by jasonmalinowski** [comment]
I think the facts are:

1. We don't know why this is happening.
2. Evidence would indicate that the problem, whatever it is, is some sort of missynced state. I believe we've had other situations recently (@Pilchie saw this I think?) where it seemed we were applying potentially stale fixes which was Bad.

I'm not against hardening this in some way, but I'd like the hardening to somehow lead us to a better place in understanding the issue. Whether that means a non-fatal Watson, some extra asserts somewhere else, etc., isn't terribly important to me, but otherwise it seems we're burying an issue here without doing due diligence. Or at least whatever due diligence has been done isn't being communicated in the commit message nor PR message.

What document ID is being passed into the functions you're hardening? The newly created document ID? And that's happening before or after TryApplyChanges?

**Comment by CyrusNajmabadi** [comment]
I've added a NFW to help out here.

**Comment by CyrusNajmabadi** [comment]
@jasonmalinowski  Please re-review.

File: src/VisualStudio/Core/Def/Utilities/IVsEditorAdaptersFactoryServiceExtensions.cs [hunk 1]
```diff
@@ -18,8 +20,22 @@ internal static class IVsEditorAdaptersFactoryServiceExtensions
             DocumentId contextDocumentId, 
             CancellationToken cancellationToken)
         {
+            // https://github.com/dotnet/roslyn/issues/17898
+            // We have a report of a null ref occuring in this method. The only place we believe 
+            // this could be would be if 'document' was null. Try to catch a reasonable 
+            // non -fatal-watson dump to help track down what the root cause of this might be.
```
Comment (jasonmalinowski) [inline src/VisualStudio/Core/Def/Utilities/IVsEditorAdaptersFactoryServiceExtensions.cs]: Space after non.

**Pull request merged**